src/components/JobLogs/withJobLogsSort.js

```diff
@@ -0,0 +1,164 @@
+  return class extends Component {
+    static propTypes = {
+      intl: intlShape.isRequired,
+      formatter: PropTypes.object,
+      resource: PropTypes.shape({
+        records: PropTypes.arrayOf(PropTypes.shape({
+          logs: PropTypes.arrayOf(PropTypes.shape({
+            fileName: PropTypes.string,
+            jobProfileName: PropTypes.string,
+            jobExecutionHrId: PropTypes.string,
+            completedDate: PropTypes.string,
+            runBy: PropTypes.shape({
+              firstName: PropTypes.string,
+              lastName: PropTypes.string,
+            }),
+          })).isRequired,
+        })),
+        isPending: PropTypes.bool.isRequired,
+      }),
+      history: PropTypes.shape({
+        push: PropTypes.func.isRequired,
+      }).isRequired,
+      location: PropTypes.shape({
+        search: PropTypes.string.isRequired,
+      }).isRequired,
+    };
+
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        sort: 'completedDate',
+        direction: 'ascending',
+      };
+
+      this.sortColumns = {
+        fileName: {
+          sortFn: sortStrings,
+          useFormatterFn: true,
+        },
+        jobProfileName: {
+          sortFn: sortStrings,
+          useFormatterFn: false,
+        },
+        jobExecutionHrId: {
+          sortFn: sortNums,
+          useFormatterFn: false,
+        },
+        runBy: {
+          sortFn: sortStrings,
+          useFormatterFn: true,
+        },
+        completedDate: {
+          sortFn: sortDates,
+          useFormatterFn: false,
+        },
+      };
+    }
+
+    componentDidMount() {
+      this.setLogsSort();
+    }
+
+    setLogsSort() {
+      const {
+        sort = 'completedDate',
+        direction = 'ascending',
+      } = qs.parse(this.props.location.search.slice(1));
+
+      this.setState({
+        sort,
+        direction,
+      });
+    }
+
+    prepareLogsData(logs = []) {
+      const {
+        sort,
+        direction,
+      } = this.state;
+
+      return logs.sort((a, b) => {
+        const cellFormatter = this.props.formatter[sort];
+        const sortDir = direction === 'ascending' ? -1 : 1;
+        const {
+          useFormatterFn,
+          sortFn,
+        } = this.sortColumns[sort];
+
+        return sortDir * (useFormatterFn && cellFormatter ?
+          sortFn(cellFormatter(a), cellFormatter(b)) :
+          sortFn(a[sort], b[sort]));
+      });
+    }
+
+    onSort = (e, { name: fieldName }) => {
+      const {
+        sort,
+        direction,
+      } = this.state;
+      const {
+        history,
+        location,
+      } = this.props;
+
+      const isSameField = sort === fieldName;
+      let sortDir = 'ascending';
+
+      if (isSameField) {
+        sortDir = direction === sortDir ? 'descending' : sortDir;
+      }
+
+      const sortState = {
+        sort: fieldName,
+        direction: sortDir,
+      };
+
+      this.setState(() => sortState);
```

Comment:
`this.setState(sortState)`?

Reply:
I did this way intentionally. I prefer this way of setting state.

https://medium.com/@wisecobbler/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1